src/Kliva/ViewModels/MainViewModel.cs

```diff
@@ -95,6 +95,34 @@ public ActivitySummary SelectedActivity
         private RelayCommand _settingsCommand;
         public RelayCommand SettingsCommand => _settingsCommand ?? (_settingsCommand = new RelayCommand(() => NavigationService.Navigate<SettingsPage>()));
 
+        private RelayCommand<ActivitySummary> _kudosCommand;
+        public RelayCommand<ActivitySummary> KudosCommand => _kudosCommand ?? (_kudosCommand = new RelayCommand<ActivitySummary>(async (x) => await OnKudosAsync(x)));
+
+        private RelayCommand<ActivitySummary> _commentCommand;
+        public RelayCommand<ActivitySummary> CommentCommand => _commentCommand ?? (_commentCommand = new RelayCommand<ActivitySummary>(async (x) => await OnCommentAsync(x)));
+
+        private async Task OnKudosAsync(ActivitySummary activitySummary)
+        {
+            // make sure we work with the swiped item
+            // this could happen if we use the mouse to swipe and move the mouse outside the ListView and release it there
+            if (activitySummary != this.SelectedActivity)
```

Comment:
meant the code this.SelectedActivity should simply be SelectedActivity, sorry for inclarity :)
I'm quite ocd on unused/unnecessary code ;)